semantic_conventions/http-common.yaml

```
@@ -5,10 +5,58 @@ groups:
     prefix: http
     attributes:
       - id: request.method
-        type: string
+        type:
+          allow_custom_values: true
+          members:
+            - id: connect
+              value: "CONNECT"
+              brief: 'CONNECT method.'
+            - id: delete
+              value: "DELETE"
+              brief: 'DELETE method.'
+            - id: get
+              value: "GET"
+              brief: 'GET method.'
+            - id: head
+              value: "HEAD"
+              brief: 'HEAD method.'
+            - id: options
+              value: "OPTIONS"
+              brief: 'OPTIONS method.'
+            - id: patch
+              value: "PATCH"
+              brief: 'PATCH method.'
+            - id: post
+              value: "POST"
+              brief: 'POST method.'
+            - id: put
+              value: "PUT"
+              brief: 'PUT method.'
+            - id: trace
+              value: "TRACE"
+              brief: 'TRACE method.'
+            - id: other
+              value: "_OTHER"
+              brief: 'Any HTTP method that the instrumentation has no prior knowledge of.'
         requirement_level: required
         brief: 'HTTP request method.'
         examples: ["GET", "POST", "HEAD"]
+        note: |
+          HTTP request method value SHOULD be known to the instrumentation.
+          By default, this convention defines "known" methods as the ones listed in [RFC9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-methods)
+          and the PATCH method defined in [RFC5789](https://www.rfc-editor.org/rfc/rfc5789.html).
+
+          If the HTTP request method is not known to instrumentation, it MUST set the `http.request.method` attribute to `_OTHER` and, if it reports spans, MUST
+          populate the exact method passed in the request line on `http.request.method_original` span attribute.
```

What about logging? SHOULD/MUST/SHOULD NOT? My impression is that rather than traces, metrics are the outlier and the MUST should apply to all signals except metrics.

CC @epixa (cf. https://github.com/open-telemetry/semantic-conventions/pull/17#issuecomment-1570821097)

```suggestion
          If the HTTP request method is not known to instrumentation, it MUST set the `http.request.method` attribute to `_OTHER` and, if except if reporting a metric MUST
          set the exact method received in the request line as value of the `http.request.method_original` attribute.
```

---

I applied your suggestion to capture the direction we think the common attribute definition should go in. 
If we ever introduce HTTP logs semconv, we'll be able to override notes in log-specific reference to the attribute and can still change the normative language there.